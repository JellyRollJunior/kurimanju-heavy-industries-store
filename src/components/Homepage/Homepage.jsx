import hero from '../../assets/hero.png';
import logo from '../../assets/logo.svg';
import styles from './Homepage.module.css';

const Homepage = () => {
  return (
    <div className={styles.layout}>
      <nav className={styles.nav}>
        <img src={logo} alt="" className={styles.logo}/>
        <button className={styles.button}>TAKE CONTROL</button>
      </nav>
      <img src={hero} alt="Mecha schematics" className={styles.hero} />
    </div>
  );
};

export { Homepage };
