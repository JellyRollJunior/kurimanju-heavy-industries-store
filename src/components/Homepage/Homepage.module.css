.layout {
  composes: primaryBackground from '../../colors.module.css';
  display: flex;
  flex-direction: column;
  min-height: 100vh;
}

.nav {
  margin-top: 80px;
}

.logo {
  margin: auto;
}

.hero {
  width: 100%;
  margin-top: 64px;
}

.buttonHolder {
  display: flex;
  max-width: 400px;
  margin: auto;
  padding: 0 32px;
}

.button {
  composes: primaryBackground from '../../colors.module.css';
  composes: secondaryColor from '../../colors.module.css';
  flex: 1;
  margin-top: 18px;
  font-size: 1.5rem;
  border: 3px dashed;
}

@media (min-width: 760px) {
  .layout {
    display: grid;
    grid-template-columns: 1fr 1.8fr;
  }

  .nav {
    margin-top: 120px;
    translate: 24px;
  }

  .logo {
    width: 100%;
  }

  .hero {
    margin-top: 0;
    padding: 46px 46px 0 0;
  }

  .button {
    border: 5px dashed;
  }
}

@media (min-width: 1024px) {
  .button {
    margin-top: 36px;
  }
}

@media (min-width: 1440px) {
  .nav {
    margin-top: 160px;
    translate: 64px;
  }

  .button {
    margin-top: 64px;
    font-size: 2rem;
  }
}

@media (min-width: 2560px) {
  .buttonHolder {
    max-width: 600px;
  }

  .button {
    margin-top: 180px;
    font-size: 3rem;
    border: 10px dashed;
  }
}
