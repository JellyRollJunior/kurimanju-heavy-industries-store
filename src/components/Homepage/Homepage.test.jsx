import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { Homepage } from './Homepage';
import { MemoryRouter } from 'react-router-dom';

describe('Homepage component', () => {
  it('Renders Homepage with hero image, company logo, and nav button', () => {
    const { container } = render(
      <MemoryRouter>
        <Homepage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
