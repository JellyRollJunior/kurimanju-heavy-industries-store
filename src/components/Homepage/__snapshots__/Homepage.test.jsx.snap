// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage component > Renders Homepage with hero image, company logo, and nav button 1`] = `
<div>
  <div
    class="_layout_1f8969"
  >
    <nav
      class="_nav_1f8969"
    >
      <img
        alt=""
        class="_logo_1f8969"
        src="/src/assets/logo.svg"
      />
      <div
        class="_buttonHolder_1f8969"
      >
        <a
          class="_navButton_1f8969"
          data-discover="true"
          href="/storefront"
        >
          TAKE CONTROL
        </a>
      </div>
    </nav>
    <img
      alt="Mecha schematics"
      class="_hero_1f8969"
      src="/src/assets/hero.png"
    />
  </div>
</div>
`;
